vignettes/user-guide.Rmd

```diff
@@ -0,0 +1,413 @@
+  parameters[[idx]]$startValue <- parameterInputData[[idx]]$start
+}
+
+# Observed data is loaded from two different files
+# because IV data is reported in µmol/L, and PO data is reported in µg/ml
+filePath <- "tests/data/Clarithromycin_Chu_1992.xlsx"
+dataConfiguration <- createImporterConfigurationForFile(filePath = filePath)
+dataConfiguration$sheets <- "IV250"
+dataConfiguration$namingPattern <- "{Sheet}"
+observedData_IV <- loadDataSetsFromExcel(xlsFilePath = filePath, importerConfigurationOrPath = dataConfiguration)
+filePath <- "tests/data/Clarithromycin_Chu_1993.xlsx"
+dataConfiguration <- createImporterConfigurationForFile(filePath = filePath)
+dataConfiguration$sheets <- c("PO250", "PO250MD", "PO500", "PO500MD")
+dataConfiguration$namingPattern <- "{Sheet}"
+observedData_PO <- loadDataSetsFromExcel(xlsFilePath = filePath, importerConfigurationOrPath = dataConfiguration)
+observedData <- c(observedData_IV, observedData_PO)
+
+outputMappings <- vector("list", length = length(simulations))
+for (idx in seq_along(simulations)) {
+  outputMappings[[idx]] <- PIOutputMapping$new(quantity = getQuantity("Organism|PeripheralVenousBlood|Clarithromycin|Plasma (Peripheral Venous Blood)",
+                                                                      container = simulations[[idx]]
+  ))
+  outputMappings[[idx]]$addObservedDataSets(observedData[[names(simulations)[[idx]]]])
+  outputMappings[[idx]]$scaling <- "lin"
+}
+
+task <- ParameterIdentification$new(
+  simulations = simulations,
+  parameters = parameters,
+  outputMappings = outputMappings,
+  configuration = piConfiguration
+)
+```
+
+### Limitations of the local optimization algorithms
+
+The solution (catalytic activity 14.10 1/min, specific clearance 6.24 1/min, intestinal permeability 1.306e-6 dm/min, OFV 4.17) found by the BOBYQA algorithm with the default parameters is satisfactory, because the individual profiles are close to the observed data. However, it is sub-optimal, as can be evidenced by non-parabolic OFV profiles:
+
+```{r include=TRUE, eval=FALSE}
+taskResults <- task$run()
+plotOFVProfiles(task$calculateOFVProfiles())[[1]]
+```
+
+```{r include=TRUE, echo=FALSE, eval=TRUE, fig.width=6, fig.height=4, fig.align='center'}
+plotOFVProfiles(cachedClarithromycinSuboptimalProfiles)[[1]]
+```
+
+If a sub-optimal solution is not enough, we need to use grid search or a global stochastic optimization algorithm, such as differential evolution, to find a good starting approximation or a global minimum.
+
+### Exhaustive grid search
+
+Running the grid search can be a useful first step when you don't know the optimization landscape of the problem.
+
+```{r include=TRUE, eval=FALSE}
+grid <- task$gridSearch(totalEvaluations = 1000, setStartingPoint = TRUE)
+best <- dplyr::slice_min(grid, ofv, with_ties = FALSE)
+```
+
+If the `setStartingPoint` argument is set to `TRUE` (instead of default `FALSE`), the next optimization run will start from the best estimate across the grid.
```

Comment:
Add more explanations. How to use `logScaleFlag` (to me it is not clear, when "a vector of logical values indicating" should be used).

How to use `margin`? (was in one of my comments to the PR introducing the grid search - I do not understand what it is and how to use it.